ApplicationLibCode/FileInterface/RifReaderOpmRft.cpp

```diff
@@ -534,21 +530,21 @@ void RifReaderOpmRft::buildSegmentData()
 
             for ( const auto& [name, arrayType, size] : results )
             {
-                if ( ( name.find( "SEG" ) == 0 ) && m_segmentResultItemCount == 0 )
+                if ( ( name.find( "SEG" ) == 0 ) && m_segmentResultItemCount.count( wellName ) == 0 )
                 {
-                    m_segmentResultItemCount = size;
+                    m_segmentResultItemCount[wellName] = size;
                 }
-                if ( name.find( "CON" ) == 0 && m_connectionResultItemCount == 0 )
+                if ( name.find( "CON" ) == 0 && m_connectionResultItemCount.count( wellName ) == 0 )
                 {
-                    m_connectionResultItemCount = size;
+                    m_connectionResultItemCount[wellName] = size;
                 }
             }
 
             for ( const auto& rftResultMetaData : results )
             {
                 const auto& [name, arrayType, size] = rftResultMetaData;
-                if ( size == static_cast<int64_t>( m_segmentResultItemCount ) ||
-                     size == static_cast<int64_t>( m_connectionResultItemCount ) )
+                if ( size == static_cast<int64_t>( m_segmentResultItemCount[wellName] ) ||
+                     size == static_cast<int64_t>( m_connectionResultItemCount[wellName] ) )
```

Comment:
These statements will insert `wellName` into `m_segmentResultItemCount` and `m_connectionResultItemCount` (and return `std::size_t{0}`).  Is that the proper semantics?

Reply:
No, this was not intended. Thanks for the feedback.